src/grdedit.c

```diff
@@ -278,6 +278,13 @@ int GMT_grdedit (void *V_API, int mode, void *args) {
 
 	/*---------------------------- This is the grdedit main code ----------------------------*/
 
+	if ((do_J == GMT->common.J.active)) {	/* Only gave -J to set the proj4 flag, so grid is already projected (i.e., Cartesian) */
+		projstring = gmt_export2proj4 (GMT);	/* Convert the GMT -J<...> into a proj4 string */
+		gmt_set_cartesian (GMT, GMT_IN);	/* Force Cartesian since processing -J changed that */
```

Comment:
Nope. We also may want to explicitly set -J"+proj=longlat ..."

Reply:
Yes, meant assignment.  I can exclude the longlat case from setting Cartesian; that is the only case no?

Reply:
From what remember now, yes. But notice that we can also say +proj=latlong